src/nupic/encoders/ScalarSensor.cpp

```diff
@@ -94,7 +94,12 @@ namespace nupic
   void ScalarSensor::compute()
   {
     Real32* array = (Real32*)encodedOutput_->getData().getBuffer();
-    const Int32 iBucket = encoder_->encodeIntoArray(sensedValue_, array);
+    UInt uintArray[encoder_->getOutputWidth()];
+    for(UInt i=0; i<encoder_->getOutputWidth(); i++) //FIXME optimize
+    {
+      uintArray[i] = (UInt)array[i];
+    }
```

Comment:
another use for `VectorHelpers`, optimal would be if the Region could provide the data as UInt already.